src/Database/LSMTree/Internal/Lookup.hs

```diff
@@ -265,22 +257,12 @@ lookupsInBatches !hbio !n !resolveV !rs !blooms !indexes !kopsFiles !ks = assert
         , V.length rs == V.length kopsFiles
         ]
 
-{-# SPECIALIZE submitInBatches ::
-       HasBlockIO IO h
-    -> BatchSize
-    -> V.Vector (IOOp RealWorld h)
-    -> IO (VU.Vector IOResult)
-  #-}
 -- | Submit I\/O operation to the 'HasBlockIO' interface in batches.
 submitInBatches ::
-     MonadAsync m
-  => HasBlockIO m h
-  -> BatchSize
+     HasBlockIO m h
   -> V.Vector (IOOp (PrimState m) h)
   -> m (VU.Vector IOResult)
-submitInBatches !hbio !n !ioops = do
-    let batches = batchesOfN (unBatchSize n) ioops
-    VU.concat <$> forConcurrently batches (submitIO hbio)
+submitInBatches !hbio !ioops = submitIO hbio ioops
```

Comment:
I'd add a second patch that inlines this and drops the "inBatches" suffix from the `lookupsInBatches` function.